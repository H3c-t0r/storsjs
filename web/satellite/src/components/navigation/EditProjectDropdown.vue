// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="edit-project">
        <div class="edit-project__selection-area" :class="{ active: isDropdownShown }" @click.stop.prevent="toggleDropdown">
            <h1 class="edit-project__selection-area__name">{{ projectName }}</h1>
            <DotsImage class="edit-project__selection-area__image"/>
        </div>
        <div class="edit-project__dropdown" v-if="isDropdownShown" v-click-outside="closeDropdown">
            <div class="edit-project__dropdown__choice" @click.stop.prevent="onEditProjectClick">
                <EditImage/>
                <p class="edit-project__dropdown__choice__label">Edit Details</p>
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import DotsImage from '@/../static/images/navigation/dots.svg';
import EditImage from '@/../static/images/navigation/edit.svg';

import { RouteConfig } from '@/router';

@Component({
    components: {
        DotsImage,
        EditImage,
    },
})
export default class EditProjectDropdown extends Vue {
    public isDropdownShown: boolean = false;

    /**
     * Returns selected project's name.
     */
    public get projectName(): string {
        return this.$store.getters.selectedProject.name;
    }

    /**
     * Redirects to edit project details page.
     */
    public onEditProjectClick(): void {
        this.closeDropdown();
        this.$router.push(RouteConfig.EditProjectDetails.path);
    }

    /**
     * Toggles dropdown visibility.
     */
    public toggleDropdown(): void {
        this.isDropdownShown = !this.isDropdownShown;
    }

    /**
     * Closes dropdown.
     */
    public closeDropdown(): void {
        this.isDropdownShown = false;
    }
}
</script>

<style scoped lang="scss">
    .edit-project {
        font-family: 'font_regular', sans-serif;
        width: 190px;
        position: relative;
        margin: 0 0 30px 13px;

        &__selection-area {
            width: calc(100% - 25px);
            display: flex;
            align-items: center;
            justify-content: space-between;
            padding: 10px 15px 10px 10px;
            border-radius: 6px;
            cursor: pointer;

            &__name {
                font-family: 'font_bold', sans-serif;
                font-size: 18px;
                line-height: 22px;
                color: #000;
                margin: 0 5px 0 0;
                word-break: break-all;
            }

            &__image {
                min-width: 3px;
            }
        }

        &__dropdown {
            position: absolute;
            top: calc(100% + 5px);
            left: 0;
            background-color: #fff;
            box-shadow: 0 8px 34px rgba(161, 173, 185, 0.41);
            border-radius: 6px;
            padding: 5px 0;
            width: 190px;
            z-index: 1;

            &__choice {
                background-color: #fff;
                width: calc(100% - 32px);
                padding: 10px 16px;
                cursor: pointer;
                display: flex;
                align-items: center;

                &__label {
                    margin: 0 0 0 10px;
                    font-weight: 500;
                    font-size: 14px;
                    line-height: 19px;
                    color: #354049;
                }

                &:hover {
                    background-color: #f5f5f7;
                }
            }
        }
    }

    .active {
        background-color: rgba(245, 246, 250, 0.7);
    }
</style>
