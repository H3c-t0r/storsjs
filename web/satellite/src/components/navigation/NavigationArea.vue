<template>
    <div class="naNavigationContainer">
        <div class="naNavigationItemContainer" v-for="navItem in navigation" v-bind:key="navItem.label" >
            <router-link class="naLinkContainer" :to="navItem.path">
                <div class="svg" v-html="navItem.svg"></div>
                <h1>{{navItem.label}}</h1>
                <div class="naAddButton" v-if="navItem.label == 'Team'">
                    <router-link to="/team/add_new">
                        <img src="../../../static/images/dashboard/Add.png" />
                    </router-link>
                </div>
            </router-link>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import NAVIGATION_ITEMS from "@/utils/constants/navigationLinks";

@Component(
    { 
        data: function() {
            return {
                navigation: NAVIGATION_ITEMS
            }
        }
    }
)

export default class NavigationArea extends Vue {}
</script>

<style lang="scss">
    .naNavigationContainer {
        position: fixed;
        width: 20vw;
		height: 100%;
		left: 0;
		top: 10vh;
        background-color: #fff;
        padding-top: 3.5vh;
    }
    a {
        text-decoration: none;
    }
    .naNavigationItemContainer {
        height: 7.5vh;
        padding-left: 4vw;
        border-left: 3px solid transparent;
        display: flex;
        justify-content: flex-start;
        align-items: center;
    }
    .naLinkContainer {
        display: flex;
        flex-direction: row;
        justify-content: flex-start;
        align-items: center;
        h1 {
            font-family: 'montserrat_medium';
            font-size: 16px;
            line-height: 23px;
            color: #354049;
            margin-left: 15px;; 
        }
    }
    .naNavigationItemContainer:hover {
        border-left: 3px solid #2683FF;
        svg path {
            fill: #2683FF !important;
        }    
    }
    .naAddButton {
        margin-left: 4vw;
        background-color: transparent;
    }
    .naAddButton:hover {
        img {
            border-radius: 50px;
            box-shadow: 0px 4px 20px rgba(35, 121, 236, 0.4);
        }
    }
</style>
