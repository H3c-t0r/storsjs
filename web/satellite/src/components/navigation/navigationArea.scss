// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

.navigation-area {
    position: relative;
    min-width: 280px;
    max-width: 280px;
    height: 100vh;
    left: 0;
    background-color: #fff;
    padding-top: 20px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    font-family: 'font_regular';

    &__logo {
        display: flex;
        flex-direction: row;
        align-items: center;
        cursor: pointer;
        margin-bottom: 60px;
        margin-left: 60px;

        &__img {
            width: 54px;
            height: 60px;
        }

        &__text {
            margin-left: 12px;
            width: 69px;
            height: 16px;
        }
    }

    &__project-title {
        font-size: 14px;
        line-height: 20px;
        color: #505872;
        margin: 0 0 10px 65px;
        font-weight: 600;
    }

    &__resources-title,
    &__account-title {
        display: flex;
        align-items: center;
        justify-content: space-between;
        margin: 0 0 0 65px;
        padding: 25px 25px 25px 0;
        border-top: 1px solid rgba(169, 181, 193, 0.3);
        font-weight: 600;

        &__title,
        &__button {
            font-size: 14px;
            line-height: 20px;
            color: #505872;
            margin: 0;
        }

        &__button {
            cursor: pointer;

            &:hover {
                color: #5E6978;
            }
        }
    }

    &__resources-title {
        margin-top: 10px;
    }

    .custom {
        padding: 25px 25px 10px 0;
    }

    &__item-container {
        height: 52px;
        padding-left: 60px;
        border-left: 3px solid transparent;
        display: flex;
        justify-content: flex-start;
        align-items: center;

        &.router-link-active,
        &:hover {
            border-left: 3px solid #2683FF;

            .svg path:not(.white) {
                fill: #2683FF !important;
            }
        }

        &__link-container {
            display: flex;
            flex-direction: row;
            justify-content: flex-start;
            align-items: center;

            h1 {
                font-family: 'font_medium';
                font-size: 16px;
                line-height: 23px;
                color: #354049;
                margin-left: 15px;
            }

            .docs-title {
                margin-left: 20px;
            }
        }
    }

    .support-item {
        margin-bottom: 10px;
    }

    .account-item {
        margin-left: 0;

        &.router-link-active {

            h1 {
                color: #2683FF;
            }
        }
    }
}

a {
    text-decoration: none;
    outline: none;
}

.disabled {
    pointer-events:none;
    opacity:0.6;
}

.divider {
    margin: 0 0 0 65px;
    border-top: 1px solid rgba(169, 181, 193, 0.3);
}

.custom-divider {
    margin: 10px 0 0 65px;
}

@media screen and (max-width: 1024px) {
    .navigation-area {
        width: 80px;
        max-width: 80px;
        min-width: 80px;

        &__logo {
            margin-left: 20px;
        }

        &__logo__text,
        &__project-title,
        &__resources-title,
        &__account-title,
        .account-item {
            display: none;
        }

        &__item-container {
            padding-left: 26px;

            &__link-container {

                h1 {
                    display: none;
                }

                &__add-button {
                    display: none;
                }
            }
        }
    }
}

@media screen and (max-height: 800px) {
    .navigation-area {

        &__item-container {
            height: 45px;
        }
    }
}

@media screen and (max-height: 730px) {
    .navigation-area {

        &__item-container {
            height: 40px;
        }
    }
}

