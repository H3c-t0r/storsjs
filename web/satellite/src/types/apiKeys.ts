// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all apiKey-related functionality
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    get(projectId: string): Promise<ApiKey[]>;
    create(projectId: string, name: string): Promise<ApiKey>;
    delete(ids: string[]): Promise<null>;
}

/**
 * ApiKey class holds info for ApiKey entity.
 */
export class ApiKey {
    public id: string;
    public secret: string;
    public name: string;
    public createdAt: string;
    public isSelected: boolean = false;

    constructor(id: string, name: string, createdAt: string, secret: string) {
        this.id = id || '';
        this.name = name || '';
        this.createdAt = createdAt || '';
        this.secret = secret || '';

        this.isSelected = false;
    }

    public formattedName(): string {
        let name = this.name;

        if (name.length < 12) {
            return name;
        }

        return name.slice(0, 12) + '...';
    }

    public getDate(): string {
        if (!this.createdAt) {
            return '';
        }

        return new Date(this.createdAt).toLocaleDateString();
    }
}
