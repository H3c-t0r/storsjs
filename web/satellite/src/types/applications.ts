// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

import VeeamIcon from '@/assets/apps/veeam.png';
import TrueNASIcon from '@/assets/apps/truenas.png';
import SplunkIcon from '@/assets/apps/splunk.png';
import ArqIcon from '@/assets/apps/arq.png';
import ZertoIcon from '@/assets/apps/zerto.png';
import UnitrendsIcon from '@/assets/apps/unitrends.png';
import S3FSIcon from '@/assets/apps/s3fs.png';
import RucioIcon from '@/assets/apps/rucio.png';
import PhotosPlusIcon from '@/assets/apps/photos+.png';
import MSP360Icon from '@/assets/apps/msp360.png';
import LucidLinkIcon from '@/assets/apps/lucidlink.png';
import IconikIcon from '@/assets/apps/iconik.png';
import HammerspaceIcon from '@/assets/apps/hammerspace.png';
import ElementsIcon from '@/assets/apps/elements.png';
import DockerIcon from '@/assets/apps/docker.png';
import PixelfedIcon from '@/assets/apps/pixelfed.png';
import MastodonIcon from '@/assets/apps/mastodon.png';
import DataverseIcon from '@/assets/apps/dataverse.png';
import CyberDuckIcon from '@/assets/apps/cyberduck.png';
import AtempoIcon from '@/assets/apps/atempo.png';
import AcronisIcon from '@/assets/apps/acronis.png';
import MountainDuckIcon from '@/assets/apps/mountainduck.png';
import FileZillaIcon from '@/assets/apps/filezilla.svg';
import GBLabsIcon from '@/assets/apps/gblabs.png';
import AdSignalIcon from '@/assets/apps/adsignal.png';
import OwnCloudIcon from '@/assets/apps/owncloud.png';
import LivepeerIcon from '@/assets/apps/livepeer.png';
import BunnyCDNIcon from '@/assets/apps/bunnycdn.png';
import CometIcon from '@/assets/apps/comet.png';
import FastlyIcon from '@/assets/apps/fastly.jpg';
import GlobusIcon from '@/assets/apps/globus.png';
import HuggingFaceIcon from '@/assets/apps/huggingface.svg';
import KerberosIcon from '@/assets/apps/kerberos.png';

export enum AppCategory {
    All = 'All',
    FileManagement = 'File Management',
    BackupRecovery = 'Backup & Recovery',
    ContentDistribution = 'Content Distribution',
    CDN = 'CDN',
    AI = 'AI',
    Logs = 'Logs',
    SocialMedia = 'Social Media',
}

export type Application = {
    title: string
    description: string
    category: AppCategory
    src: string
    docs: string
}

export const applications: Application[] = [
    {
        title: 'TrueNAS - iX Systems',
        description: 'Back up a single instance or keep several TrueNAS deployments synchronized using a single source of truth.',
        category: AppCategory.BackupRecovery,
        src: TrueNASIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ix-systems-truenas',
    },
    {
        title: 'Nebula by GB Labs',
        description: 'Cloud-based media storage solution powered by Storj. Work directly from the GB Labs Cloud with project files that are always in sync.',
        category: AppCategory.FileManagement,
        src: GBLabsIcon,
        docs: 'https://www.storj.io/partner-solutions/gb-labs',
    },
    {
        title: 'Hammerspace',
        description: 'Hammerspace and Storj bring more flexibility and agility to your global data landscape.',
        category: AppCategory.FileManagement,
        src: HammerspaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hammerspace',
    },
    {
        title: 'Ad Signal Match',
        description: 'Deduplicate assets and save on AI costs with Ad Signal Match. Transfer only your unique content to Storj.',
        category: AppCategory.FileManagement,
        src: AdSignalIcon,
        docs: 'https://www.storj.io/partner-solutions/ad-signal',
    },
    {
        title: 'OwnCloud',
        description: 'oCIS, or ownCloud Infinite Scale, is a cutting-edge technology platform for building cloud-native file sync and share applications.',
        category: AppCategory.FileManagement,
        src: OwnCloudIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ocis',
    },
    {
        title: 'Livepeer',
        description: 'Add live and on-demand video experiences by transcoding videos with Livepeer and storing your media with Storj.',
        category: AppCategory.BackupRecovery,
        src: LivepeerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/livepeer',
    },
    {
        title: 'Acronis',
        description: 'Reliable backup and disaster recovery solutions for data archiving and organization, seamlessly integrating with Storj.',
        category: AppCategory.BackupRecovery,
        src: AcronisIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/acronis',
    },
    {
        title: 'Iconik',
        description: 'Iconik is easy to use and intuitive with a clean and intuitive platform that organizes your media and makes it searchable.',
        category: AppCategory.FileManagement,
        src: IconikIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/iconik',
    },
    {
        title: 'Photos+',
        description: 'Store and manage your photos and videos in your own cloud storage account.',
        category: AppCategory.FileManagement,
        src: PhotosPlusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/photos-plus',
    },
    {
        title: 'Veeam',
        description: 'All-in-one backup, recovery, and data security solution that serves both on-premises and cloud storage.',
        category: AppCategory.BackupRecovery,
        src: VeeamIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/veeam',
    },
    {
        title: 'Splunk',
        description: 'Splunk is a data analytics platform that provides data-driven insights across all aspects of a company.',
        category: AppCategory.Logs,
        src: SplunkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/splunk',
    },
    {
        title: 'Arq Backup',
        description: 'Arq backs up your filesystem with perfect, point-in-time backups of your files.',
        category: AppCategory.BackupRecovery,
        src: ArqIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/arq',
    },
    {
        title: 'Zerto',
        description: 'Easily incorporate immutability into a complete disaster recovery strategy.',
        category: AppCategory.BackupRecovery,
        src: ZertoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/zerto',
    },
    {
        title: 'Unitrends',
        description: 'Automate manual tasks, eliminate management complexity, and deliver tested hardware and software resilience.',
        category: AppCategory.BackupRecovery,
        src: UnitrendsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/unitrends',
    },
    {
        title: 'S3FS',
        description: 's3fs allows Linux, macOS, and FreeBSD to mount an S3 bucket via FUSE.',
        category: AppCategory.FileManagement,
        src: S3FSIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/s3fs',
    },
    {
        title: 'Rucio',
        description: 'Organize, manage, and access your data at scale with Rucio, an open-source framework for scientific collaboration.',
        category: AppCategory.FileManagement,
        src: RucioIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/rucio',
    },
    {
        title: 'MSP360',
        description: 'MSP360 is a best-in-class IT management platform for MSPs and internal IT teams.',
        category: AppCategory.BackupRecovery,
        src: MSP360Icon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/msp360',
    },
    {
        title: 'LucidLink',
        description: 'Using Storj with LucidLink provides resilient cloud object storage with blazing performance and zero-trust security.',
        category: AppCategory.FileManagement,
        src: LucidLinkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/lucidlink',
    },
    {
        title: 'Elements',
        description: 'Store, manage and collaborate on media content using a secure and scalable asset management platform.',
        category: AppCategory.FileManagement,
        src: ElementsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/elements',
    },
    {
        title: 'Docker',
        description: 'Storj supports custom Content-Type for any key, so it can be used as a container registry to distribute container images.',
        category: AppCategory.ContentDistribution,
        src: DockerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/docker',
    },
    {
        title: 'Pixelfed',
        description: 'Learn how to set up Pixelfed decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: PixelfedIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/pixelfed',
    },
    {
        title: 'Mastodon',
        description: 'Here is how you can set up Mastodon decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: MastodonIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/mastodon',
    },
    {
        title: 'Dataverse',
        description: 'S3 credentials allow Dataverse to upload and download files from Storj as if it was using the S3 API.',
        category: AppCategory.FileManagement,
        src: DataverseIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/dataverse',
    },
    {
        title: 'CyberDuck',
        description: 'Open-source cloud storage browser app for macOS, Windows, and Linux that supports Storj.',
        category: AppCategory.FileManagement,
        src: CyberDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Atempo',
        description: 'Integrate with Storj to reduce storage costs without sacrificing security or performance for media archival with global access.',
        category: AppCategory.BackupRecovery,
        src: AtempoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/atempo-miria',
    },
    {
        title: 'Mountain Duck',
        description: 'Add Storj as a disk on your computer. Open your Storj files with any application and work like on a local disk.',
        category: AppCategory.FileManagement,
        src: MountainDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Bunny CDN',
        description: 'Set up a static website hosting with Storj using Bunny CDN as a content delivery network providing a caching layer.',
        category: AppCategory.CDN,
        src: BunnyCDNIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/bunny',
    },
    {
        title: 'Fastly',
        description: 'Distribute your content among the Fastly edge cloud service using your Storj buckets as a source of content.',
        category: AppCategory.CDN,
        src: FastlyIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/fastly',
    },
    {
        title: 'Comet',
        description: 'Flexible backup platform integrates with Storj as a backup and storage destination to protect and restore data.',
        category: AppCategory.BackupRecovery,
        src: CometIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/comet',
    },
    {
        title: 'Globus',
        description: 'Seamless collaboration and management for data transfer, sharing, and discovery, all into one unified open-source platform.',
        category: AppCategory.FileManagement,
        src: GlobusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/globus',
    },
    {
        title: 'HuggingFace',
        description: 'Train and deploy open-source AI models, while saving and loading datasets to and from Storj.',
        category: AppCategory.AI,
        src: HuggingFaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hugging-face',
    },
    {
        title: 'FileZilla',
        description: 'Learn how to set up FileZilla to transfer files over Storj DCS or integrate FileZilla Pro.',
        category: AppCategory.FileManagement,
        src: FileZillaIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/filezilla',
    },
    {
        title: 'Kerberos.io',
        description: 'Open-source platform for video analytics and management. Integrate with Storj to store your Kerberos Vault video files.',
        category: AppCategory.FileManagement,
        src: KerberosIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/kerberos-vault',
    },
];
