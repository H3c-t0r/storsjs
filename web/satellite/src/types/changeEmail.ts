// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ChangeEmailStep {
    InitStep = 0,
    VerifyPasswordStep,
    Verify2faStep,
    VerifyOldEmailStep,
    SetNewEmailStep,
    VerifyNewEmailStep,
    SuccessStep,
}
