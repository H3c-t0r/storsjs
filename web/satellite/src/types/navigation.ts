// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export class NavigationLink {
    private _path: string;
    private _name: string;
    private _icon: string;

    public constructor(path: string, name: string, icon: string = '') {
        this._path = path;
        this._name = name;
        this._icon = icon;
    }

    public get path(): string {
        return this._path;
    }
    public get name(): string {
        return this._name;
    }
    public get icon(): string {
        return this._icon;
    }

    public isChild(): boolean {
        return this._path[0] !== '/';
    }

    public withIcon(icon: string): NavigationLink {
        return new NavigationLink(this._path, this._name, icon);
    }

    public with(child: NavigationLink): NavigationLink {
        if (!child.isChild()) {
            // TODO: better error message
            throw new Error('child root is not child');
        }

        return new NavigationLink(`${this.path}/${child.path}`, child.name);
    }
}
