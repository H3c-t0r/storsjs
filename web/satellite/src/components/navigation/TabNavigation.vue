// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="tab-navigation-container">
        <router-link
            :to="navLink.path"
            class="tab-navigation-container__item"
            v-for="navLink in navigation"
            :key="navLink.name">
            <p class="tab-navigation-container__item__name">{{navLink.name}}</p>
        </router-link>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { NavigationLink } from '@/types/navigation';

@Component
export default class TabNavigation extends Vue {
    @Prop({default: new Array(NavigationLink)})
    private navigation: NavigationLink[];
}
</script>

<style scoped lang="scss">
    .tab-navigation-container {
        width: auto;
        display: flex;
        border-bottom: 1px solid #afb7c1;

        &__item {
            width: 150px;
            height: 48px;
            display: flex;
            align-items: center;
            justify-content: center;

            &__name {
                font-family: 'font_medium', sans-serif;
                font-size: 14px;
                line-height: 20px;
                color: #afb7c1;
            }

            &.router-link-exact-active,
            &:hover {
                border-bottom: 3px solid #2683ff;

                .tab-navigation-container__item__name {
                    color: #354049;
                }
            }
        }
    }
</style>
