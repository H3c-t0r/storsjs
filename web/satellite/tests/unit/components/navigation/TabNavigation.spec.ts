// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import VueRouter from 'vue-router';

import TabNavigation from '@/components/navigation/TabNavigation.vue';

import { NavigationLink } from '@/types/navigation';
import { createLocalVue, mount, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();

localVue.use(VueRouter);

const navigation: NavigationLink[] = [
    new NavigationLink('path1', 'name1'),
    new NavigationLink('path2', 'name2'),
    new NavigationLink('path3', 'name3'),
];

describe('TabNavigation', () => {
    it('snapshot not changed', () => {
        const wrapper = shallowMount(TabNavigation, {
            localVue,
            propsData: {
                navigation,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('navigation links renders correctly', () => {
        const wrapper = shallowMount(TabNavigation, {
            localVue,
            propsData: {
                navigation,
            },
        });

        const navigationLinks = wrapper.findAll('.tab-navigation-container__item');

        expect(navigationLinks.length).toBe(navigation.length);

        for (let i = 0; i < navigation.length; i++) {
            expect(navigationLinks.at(i).text()).toBe(navigation[i].name);
        }
    });
});
