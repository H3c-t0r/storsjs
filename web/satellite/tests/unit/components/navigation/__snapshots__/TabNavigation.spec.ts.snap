// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNavigation snapshot not changed 1`] = `
<div class="tab-navigation-container">
  <router-link-stub to="path1" tag="a" event="click" class="tab-navigation-container__item">
    <p class="tab-navigation-container__item__name">name1</p>
  </router-link-stub>
  <router-link-stub to="path2" tag="a" event="click" class="tab-navigation-container__item">
    <p class="tab-navigation-container__item__name">name2</p>
  </router-link-stub>
  <router-link-stub to="path3" tag="a" event="click" class="tab-navigation-container__item">
    <p class="tab-navigation-container__item__name">name3</p>
  </router-link-stub>
</div>
`;
